### ISSUE TITLE
model

### ISSUE THREAD

**Issue by congcongzhang1996** [ytc272098215/HGAT#18]
Is the model in the code the same with the description in the paper? I have some doubt on the (models.py).

**Comment by abhinab303** [comment]
Have you understood the code now? It seems that in paper, type-level attention is calculated followed by node-level. But in code it looks quite the opposite. Also, in node-level attention, there is no concatenation operation like in paper. And even after applying softmax, there are extra steps, I don't understand what it has to do with the equations in the paper.

```
attention = F.softmax(attention, dim=1)
attention = torch.mul(attention, adj.sum(1).repeat(M, 1).t())
attention = torch.add(attention * self.gamma, adj.to_dense() * (1 - self.gamma))
h_prime = torch.matmul(attention, g)
```
